### ISSUE TITLE
MySQL Error: Error in migration engine

### ISSUE THREAD

**Issue by marktopper** [prisma/prisma-examples#865]
I did a clean setup of Prisma2 with the following input/output:
```
$ prisma2 init prisma2-mysql
Languages for starter kits: Starter Kit
Photon is available in these languages: TypeScript
Available starter kits: GraphQL API (+ Auth)
Supported databases: MySQL
MySQL Url: mysql://user:pass@localhost:3306/prisma
Database options: Create MySQL database `prisma`

Preparing your starter kit: GraphQL API (+ Auth)

✔  Downloading the starter kit from GitHub ...
✔  Extracting content to prisma2-mysql ...
✔  Installing dependencies with: `npm install` ...
✔  Preparing your database ...
✔  Seeding your database with: `npm run seed` ... 


 SUCCESS  The prisma2-mysql directory was created!
 SUCCESS  Prisma is connected to your database at localhost

┌─ Next steps │
│
│   Navigate into the project directory:
│
│   $ cd prisma2-mysql
│
│   Start Prisma's development mode to enable access to
│   Prisma Studio and watch schema.prisma for changes:
│   $ prisma2 dev
│
│   Start the GraphQL server (in a new terminal window/tab):
│   $ npm run dev
│
│   Learn more about Prisma 2:
│   https://github.com/prisma/prisma2/
│
│   If you encounter any issues, please report them here:
│   https://github.com/prisma/prisma-examples/issues/new
│
└─

$ cd prisma2-mysql
$ prisma2 dev


Error

Error: Error in migration engine: [migration-engine/connectors/sql-migration-connector/src/sql_migration_persistence.rs
:40] &sql_str = "CREATE TABLE `prisma`.`_Migration` (`revision` INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY, `name` TEX
T NOT NULL, `datamodel` LONGTEXT NOT NULL, `status` TEXT NOT NULL, `applied` INTEGER NOT NULL, `rolled_back` INTEGER NO
T NULL, `datamodel_steps` LONGTEXT NOT NULL, `database_migration` LONGTEXT NOT NULL, `errors` LONGTEXT NOT NULL, `start
ed_at` datetime(3) NOT NULL, `finished_at` datetime(3));"
thread 'tokio-runtime-worker-0' panicked at 'get data_type', src/libcore/option.rs:1190:5
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: migration_engine::main::{{closure}}
   6: std::panicking::rust_panic_with_hook
   7: std::panicking::continue_panic_fmt
   8: rust_begin_unwind
   9: core::panicking::panic_fmt
  10: core::option::expect_failed
  11: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &mut F>::call_once
  12: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter
  13: sql_schema_describer::mysql::SqlSchemaDescriber::get_table
  14: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
  15: <sql_schema_describer::mysql::SqlSchemaDescriber as sql_schema_describer::SqlSchemaDescriberBackend>::describe
  16: <sql_migration_connector::sql_database_migration_inferrer::SqlDatabaseMigrationInferrer as migration_connector::d
atabase_migration_inferrer::DatabaseMigrationInferrer<sql_migration_connector::sql_migration::SqlMigration>>::infer
  17: <migration_engine::commands::infer_migration_steps::InferMigrationStepsCommand as migration_engine::commands::com
mand::MigrationCommand>::execute
  18: <migration_engine::api::MigrationApi<C,D> as migration_engine::api::GenericApi>::infer_migration_steps
  19: migration_engine::api::rpc::RpcApi::create_sync_handler
  20: tokio_executor::enter::exit
  21: tokio_threadpool::blocking::blocking
  22: <futures::future::lazy::Lazy<F,R> as futures::future::Future>::poll
  23: futures::future::chain::Chain<A,B,C>::poll
  24: <futures::future::then::Then<A,B,F> as futures::future::Future>::poll
  25: <futures::future::lazy::Lazy<F,R> as futures::future::Future>::poll
  26: <futures::future::then::Then<A,B,F> as futures::future::Future>::poll
  27: <futures::future::map::Map<A,F> as futures::future::Future>::poll
  28: <futures::future::either::Either<A,B> as futures::future::Future>::poll
  29: <futures::future::map::Map<A,F> as futures::future::Future>::poll
  30: <futures::future::map_err::MapErr<A,F> as futures::future::Future>::poll
  31: <futures::stream::and_then::AndThen<S,F,U> as futures::stream::Stream>::poll
  32: <futures::stream::forward::Forward<T,U> as futures::future::Future>::poll
  33: <futures::future::map::Map<A,F> as futures::future::Future>::poll
  34: <futures::future::map_err::MapErr<A,F> as futures::future::Future>::poll
  35: futures::task_impl::std::set
  36: std::panicking::try::do_call
  37: __rust_maybe_catch_panic
  38: tokio_threadpool::task::Task::run
  39: tokio_threadpool::worker::Worker::run_task
  40: tokio_threadpool::worker::Worker::run
  41: std::thread::local::LocalKey<T>::with
  42: std::thread::local::LocalKey<T>::with
  43: std::thread::local::LocalKey<T>::with
  44: tokio::runtime::threadpool::builder::Builder::build::{{closure}}
  45: std::thread::local::LocalKey<T>::with
  46: std::thread::local::LocalKey<T>::with
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
    at ChildProcess.<anonymous> (/usr/local/lib/node_modules/prisma2/build/index.js:215297:35)
    at ChildProcess.emit (events.js:209:13)
    at ChildProcess.EventEmitter.emit (domain.js:476:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
```

This error does not happen when do the setup with `sqlite`.

**Comment by tomhoule** [comment]
Thanks for reporting the bug! This is a  known problem that we worked on, a fix was merged this week and it will be in the next preview version.

Relevant issues: https://github.com/prisma/prisma-engine/issues/49 and https://github.com/prisma/lift/issues/107

This is a problem that only occurs on MySQL 8, so as a temporary workaround until the next preview, using MySQL 5.7 should work. I hope this helps :)
